test/e2e/catalog_e2e_test.go

```diff
@@ -1309,6 +1309,179 @@ var _ = Describe("Starting CatalogSource e2e tests", func() {
 			}
 		}
 	})
+
+	When("A CatalogSource is created with a properly formed CSV", func() {
+		var (
+			og               operatorsv1.OperatorGroup
+			namespaceName    string
+			packageName      string
+			stableChannel    string
+			subscriptionName string
+			manifests        []registry.PackageManifest
+			crd              apiextensions.CustomResourceDefinition
+			csv              v1alpha1.ClusterServiceVersion
+			cs               *v1alpha1.CatalogSource
+		)
+
+		BeforeEach(func() {
+			namespaceName = genName("good-csv-")
+			og = operatorsv1.OperatorGroup{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fmt.Sprintf("%s-operatorgroup", namespaceName),
+					Namespace: namespaceName,
+				},
+			}
+			ns = SetupGeneratedTestNamespaceWithOperatorGroup(namespaceName, og)
+
+			// create a simple catalogsource
+			packageName = genName("good-csv-")
+			stableChannel = "stable"
+			packageStable := packageName + "-stable"
+			manifests = []registry.PackageManifest{
+				{
+					PackageName: packageName,
+					Channels: []registry.PackageChannel{
+						{Name: stableChannel, CurrentCSVName: packageStable},
+					},
+					DefaultChannelName: stableChannel,
+				},
+			}
+
+			crd = newCRD(genName("ins"))
+			csv = newCSV(packageStable, ns.GetName(), "", semver.MustParse("0.1.0"), []apiextensions.CustomResourceDefinition{crd}, nil, nil)
+
+			catalogSourceName := genName("mock-ocs-")
+			cs, _ = createInternalCatalogSource(c, crc, catalogSourceName, ns.GetName(), manifests, []apiextensions.CustomResourceDefinition{crd}, []v1alpha1.ClusterServiceVersion{csv})
+			_, err := fetchCatalogSourceOnStatus(crc, catalogSourceName, ns.GetName(), catalogSourceRegistryPodSynced)
+			Expect(err).ShouldNot(HaveOccurred())
```

Comment:
@perdasilva Should we start introducing the magic catalog implementation here? We don't have to refactor everything to use that library, but we could enforce all net new e2e tests use that package to bootstrap catalogs for testing purposes.

Reply:
I'm totally good with that! I think it makes things much easier and leaner on the code side.

Reply:
When rewriting these tests with the magic catalog is the happy path (everything properly installs from a properly formed csv) need to re-written? I might be misunderstanding but it seems like that's already tested in the main magic-catalog tests [here](https://github.com/operator-framework/operator-lifecycle-manager/blob/604825027920cdc52a489669b22f51910cabfa69/test/e2e/magic_catalog_test.go#L29).